platform/flowglad-next/src/components/forms/CreatePriceModal.tsx

```diff
@@ -3,43 +3,96 @@
 import { trpc } from '@/app/_trpc/client'
 import FormModal from '@/components/forms/FormModal'
 import { useAuthenticatedContext } from '@/contexts/authContext'
-import { createPriceFormSchema } from '@/db/schema/prices'
+import { createPriceFormSchema, Price } from '@/db/schema/prices'
 import { PriceType } from '@/types'
-import { rawStringAmountToCountableCurrencyAmount } from '@/utils/stripe'
+import {
+  countableCurrencyAmountToRawStringAmount,
+  rawStringAmountToCountableCurrencyAmount,
+} from '@/utils/stripe'
 import PriceFormFields from './PriceFormFields'
 
 interface CreatePriceModalProps {
   isOpen: boolean
   setIsOpen: (isOpen: boolean) => void
   productId: string
+  previousPrice: Price.ClientRecord
+}
+
+const getDefaultValues = (
+  previousPrice: Price.ClientRecord
+): { price: Price.ClientInsert; __rawPriceString: string } => {
+  const coreValues = {
+    productId: previousPrice.productId,
+    isDefault: previousPrice.isDefault,
+    unitPrice: previousPrice.unitPrice,
+    slug: previousPrice.slug,
+    active: true,
+  }
+
+  const __rawPriceString = countableCurrencyAmountToRawStringAmount(
+    previousPrice.currency,
+    previousPrice.unitPrice!
```

Comment:
non-null assertion doesn't seem necessary
```suggestion
    previousPrice.unitPrice
```